libp2p/varint.nim

```diff
@@ -309,84 +312,87 @@ proc putSVarint*(pbytes: var openarray[byte], outsize: var int,
   ## Maximum encoded length of 64bit integer is 10 octets.
   ## Maximum encoded length of 32bit integer is 5 octets.
   when sizeof(outval) == 8:
-    result = PB.putUVarint(pbytes, outsize, uint64(outval))
+    PB.putUVarint(pbytes, outsize, uint64(outval))
   else:
-    result = PB.putUVarint(pbytes, outsize, uint32(outval))
+    PB.putUVarint(pbytes, outsize, uint32(outval))
 
 template varintFatal(msg) =
   const m = msg
   {.fatal: m.}
 
 proc putVarint*[T: PB|LP](vtype: typedesc[T], pbytes: var openarray[byte],
-                nbytes: var int, value: SomeVarint): VarintStatus {.inline.} =
+                nbytes: var int, value: SomeVarint): VarintResult[void] {.inline.} =
   when vtype is PB:
     when (type(value) is PBSomeSVarint) or (type(value) is PBZigVarint):
-      result = putSVarint(pbytes, nbytes, value)
+      putSVarint(pbytes, nbytes, value)
     elif (type(value) is PBSomeUVarint):
-      result = PB.putUVarint(pbytes, nbytes, value)
+      PB.putUVarint(pbytes, nbytes, value)
     else:
       varintFatal("Protobuf's varint do not support type [" &
                   typetraits.name(type(value)) & "]")
   elif vtype is LP:
     when (type(value) is LPSomeVarint):
-      result = LP.putUVarint(pbytes, nbytes, value)
+      LP.putUVarint(pbytes, nbytes, value)
     else:
       varintFatal("LibP2P's varint do not support type [" &
                    typetraits.name(type(value)) & "]")
 
 proc getVarint*[T: PB|LP](vtype: typedesc[T], pbytes: openarray[byte],
                           nbytes: var int,
-                          value: var SomeVarint): VarintStatus {.inline.} =
+                          value: var SomeVarint): VarintResult[void] {.inline.} =
   when vtype is PB:
     when (type(value) is PBSomeSVarint) or (type(value) is PBZigVarint):
-      result = getSVarint(pbytes, nbytes, value)
+      getSVarint(pbytes, nbytes, value)
     elif (type(value) is PBSomeUVarint):
-      result = PB.getUVarint(pbytes, nbytes, value)
+      PB.getUVarint(pbytes, nbytes, value)
     else:
       varintFatal("Protobuf's varint do not support type [" &
                   typetraits.name(type(value)) & "]")
   elif vtype is LP:
     when (type(value) is LPSomeVarint):
-      result = LP.getUVarint(pbytes, nbytes, value)
+      LP.getUVarint(pbytes, nbytes, value)
     else:
       varintFatal("LibP2P's varint do not support type [" &
                    typetraits.name(type(value)) & "]")
 
 proc encodeVarint*(vtype: typedesc[PB],
-                   value: PBSomeVarint): seq[byte] {.inline.} =
+                   value: PBSomeVarint): VarintResult[seq[byte]] {.inline.} =
```

Comment:
encoding a varint here can't fail - pb varints are defined for all inputs - this should probably be addressed at some point

Reply:
what do you suggest? like this ok for now?
Was throwing before :)

Reply:
yeah ok for now - it's a bit annoying in that it causes unnecessary error handling downstream, but because of LP, the error has to be there to support same api